src/object-pool-options.ts

```diff
@@ -1,3 +1,6 @@
-export interface ArmorObjectPoolOptions {
+import ArmorObjectPoolState from './object-pool-state';
 
+export default interface ArmorObjectPoolOptions<T> {
+	state?: ArmorObjectPoolState<T> | string;
```

Comment:
I was thinking about our discussion here and think having a state property could be ambiguous. We can clarify somewhat by changing the name from `state` to `serializedState` and remove the type `ArmorObjectPoolState<T>` making it only `string`.

Reply:
Also note this change will obviously need to be made also throughout the file.